wwwapp/auth.py

```diff
@@ -41,154 +32,74 @@ def login_view(request):
-        # previous provider, connect the new provider.
-        if 'merge_access' in self.request.session:
-            pk = self.request.session['merge_access']
-            AccountAccess.objects.filter(pk=pk).update(user=user)
-            del self.request.session['merge_access']
-            del self.request.session['merge_access_info']
-        login(self.request, user)
-        return redirect(self.get_login_redirect(provider, user, access))
-
-    def get_or_create_user(self, provider, access, info):
-        return create_user(access, info)
-
-
-# Creates a django User with a nice username from info.
-def create_user(access, info):
-    username = ''
-    if 'last_name' in info:
-        name = re.sub(r'[^a-zA-Z0-9]', '', info['last_name'])
-        username = name[:18]
-    if 'first_name' in info:
-        name = re.sub(r'[^a-zA-Z0-9]', '', info['first_name'])
-        l = 22 - len(username)
-        username = name[:l] + username
-    if len(username) < 5 and 'email' in info:
-        name = re.sub(r'@.*$', '', info['email'])
-        name = re.sub(r'[^a-zA-Z0-9]', '', name)
-        l = 22 - len(username)
-        username = name[:l] + username
-
-    digest = hashlib.sha1(smart_bytes(access)).digest()
-    digest = force_text(base64.urlsafe_b64encode(digest)).replace('=', '')
-    username = username + '-' + digest[-7:]
-
-    kwargs = {
-        User.USERNAME_FIELD: username,
-        'email': '',
-        'password': None
-    }
-    if 'email' in info:
-        kwargs['email'] = info['email']
-    if 'first_name' in info:
-        kwargs['first_name'] = info['first_name']
-    if 'last_name' in info:
-        kwargs['last_name'] = info['last_name']
-
-    return User.objects.create_user(**kwargs)
-
-
-# The view called when a user decides not to merge into any suggested accounts.
-def create_user_from_unmerged_access_view(request):
-    if 'merge_access' not in request.session:
-        raise ValidationError('No AccountAccess to create User from.')
-    pk = request.session['merge_access']
-    info = request.session['merge_access_info']
-    access = AccountAccess.objects.get(pk=pk)
-    user = create_user(access, info)
-    access.user = user
-    AccountAccess.objects.filter(pk=pk).update(user=user)
-    del request.session['merge_access']
-    del request.session['merge_access_info']
-    user = authenticate(provider=access.provider, identifier=access.identifier)
-    login(request, user)
-    return redirect(settings.LOGIN_REDIRECT_URL)
+            return render(request, 'loginMerge.html', context)
+
 
+def resume_partial(request, provider):
+    request.session['partial_pipeline_token'] = request.GET.get('partial_token')
+    request.session['next'] = '/'
+    if request.user.is_authenticated:
+        return redirect(reverse('social:complete', args=(provider,)))
+    else:
+        return redirect(reverse('social:complete', args=(provider,)) + "?new_account=1")
```

Comment:
This has a risk of somebody modifying the request.... :man_facepalming:

Reply:
Also don't specify request params manually. Add a parameter to reverse instead.